zebra-test/src/command.rs

```diff
@@ -9,12 +9,29 @@ use tempdir::TempDir;
 #[cfg(unix)]
 use std::os::unix::process::ExitStatusExt;
 
+/// Config file to change default addresses and ports
+static TEST_CONFIG: &str = "[metrics]
+endpoint_addr = '127.0.0.1:0'
+[network]
+listen_addr = '127.0.0.1:0'
+[tracing]
+endpoint_addr = '127.0.0.1:0'
+";
```

Comment:
Rather than creating a static toml string lets make a function to generate a `Config` with `Config::default()` followed by overriding the fields we care about. Then we can write that to the filesystem using serde and serde_toml.

Reply:
As part of this we might want to take the logic in `zebrad generate` and move it into helper functions on the `Config` type that we can reuse in our test setup code and in the generate command.

Reply:
We could pass an optional comment to the function that writes the config?

Reply:
yea, but I don't think doing so accomplishes anything meaningful. This isn't code where drift between the various impls will cause problems, writing to a file is pretty universal.